### ISSUE TITLE
[Android] Crash when parsing JS stacktrace

### ISSUE THREAD

**Issue by haggholm** [facebook/react-native#13286]
### Description

Very similar to  #8311, but rather than *no* stack, some of stack information is missing.

```
com.facebook.react.bridge.UnexpectedNativeTypeException: TypeError: expected dynamic type `int64', but had type `null'
  at com.facebook.react.bridge.ReadableNativeMap.getInt(Native Method)
  at com.facebook.react.devsupport.StackTraceHelper.convertJsStackTrace(StackTraceHelper.java:103)
  at com.facebook.react.devsupport.DevSupportManagerImpl$3.run(DevSupportManagerImpl.java:287)
  at android.os.Handler.handleCallback(Handler.java:739)
  at android.os.Handler.dispatchMessage(Handler.java:95)
  at android.os.Looper.loop(Looper.java:148)
  at android.app.ActivityThread.main(ActivityThread.java:5417)
  at java.lang.reflect.Method.invoke(Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

which points me to

```java
  public static StackFrame[] convertJsStackTrace(@Nullable ReadableArray stack) {
    int size = stack != null ? stack.size() : 0;
    StackFrame[] result = new StackFrame[size];
    for (int i = 0; i < size; i++) {
      ReadableMap frame = stack.getMap(i);
      String methodName = frame.getString("methodName");
      String fileName = frame.getString("file");
      int lineNumber = frame.getInt("lineNumber");    // <-- Crashes here
      int columnNumber = -1;
      if (frame.hasKey("column") && !frame.isNull("column")) {
        columnNumber = frame.getInt("column");
      }
      result[i] = new StackFrame(fileName, methodName, lineNumber, columnNumber);
    }
    return result;
  }
```

### Solution

Unfortunately, I don’t immediately know if I can provide the code that caused the issue (I’d have to check up on some policies). However, maybe it could be so simple as to have explicit checks for the presence of the fields and/or changing to nullable object types rather than primitive `int`s?

### Additional Information

* React Native version: 0.42.3
* Platform: Android (have not tested iOS)
* Development Operating System: Linux
* Dev tools: Android Studio 2.3, Android 6.0

**Comment by haggholm** [comment]
By a charming coincidence, this was fixed by another Petter H (@petterh), in https://github.com/facebook/react-native/commit/957b55c769661edeee691109e45dc99a06f87bc4; but it is only available in v0.44.0-rc.0. It’s not straightforward to upgrade to this version, since it requires an unreleased version of React. I wonder if there’s any chance of backporting?

**Comment by hramos** [comment]
It's unlikely this commit will be backported into 0.43, as the stable release has already been cut. You should still be able to use the RC version, we actually encourage people to use the RC in order to surface these kinds of issues as soon as possible.
